### PR TITLE
AAD sample and README revised, and correct indentation

### PR DESCRIPTION
**Pull request by mssfang**
fixes:  #6884
fixes: #7089 Add java code to README by using embedme plugin:

### CODE REVIEW

File: sdk/appconfiguration/azure-data-appconfiguration/README.md [hunk 1]
```diff
@@ -135,14 +137,60 @@ ConfigurationAsyncClient client = new ConfigurationClientBuilder()
...
+[DefaultAzureCredential][default_cred_ref] will be able to authenticate the
+configuration client.
+
+Constructing the client also requires your configuration store's URL, which you can
+get from the Azure CLI or the Azure Portal. In the Azure Portal, the URL can be found listed as the service "Endpoint"
 
 ```Java
-// An example of using TokenCredential and Endpoint to create a synchronous client
-TokenCredential credential = new DefaultAzureCredential();
+import com.azure.identity.DefaultAzureCredential;
```
Comment (srnagar) [inline sdk/appconfiguration/azure-data-appconfiguration/README.md]: Prefer to use code embedding for READMEs.

Comment (mssfang) [reply sdk/appconfiguration/azure-data-appconfiguration/README.md]: updated.

File: sdk/appconfiguration/azure-data-appconfiguration/README.md [hunk 2]
```diff
@@ -223,14 +223,15 @@ String url = configurationClient.getConfigurationSetting(urlKey, urlLabel).getVa
 Connection conn;
 try {
     conn = DriverManager.getConnection(url);
+    conn.close();
```
Comment (conniey) [inline sdk/appconfiguration/azure-data-appconfiguration/README.md]: I'd put this in a try-finally block, so if any errors happen, it'll always try to dispose of the connection.
```java
try {
} finally {
    if (conn != null) {
        conn.close();
    }
}
```

Comment (mssfang) [reply sdk/appconfiguration/azure-data-appconfiguration/README.md]: but how do you handle the exception? Ignore the exception and throws it? like the example below: 
```
 public void sqlExample() throws SQLException {
        ConfigurationClient configurationClient = new ConfigurationClientBuilder()
            .connectionString(connectionString)
            .buildClient();

        // urlLabel is optional
        String url = configurationClient.getConfigurationSetting(urlKey, urlLabel).getValue();
        Connection conn = null;
        try {
            conn = DriverManager.getConnection(url);
        } finally {
            if (conn != null) {
                conn.close();
            }
        }
    }

```

Comment (conniey) [reply sdk/appconfiguration/azure-data-appconfiguration/README.md]: Try finally can have a catch block too.
https://www.javatpoint.com/finally-block-in-exception-handling

Comment (mssfang) [reply sdk/appconfiguration/azure-data-appconfiguration/README.md]: conn.close() throws exception as well

**Pull request merged**